Automattic/jetpack
# Add @automattic/jetpack-analytics package

related: https://github.com/Automattic/jetpack/pull/20543

Moves most analytics-related JS into the `@automattic/jetpack-analytics`.

This was extracted out of the [Standalone My Plans Page PR](https://github.com/Automattic/jetpack/pull/20543).

The benefit of a standalone analytics package is that we can have components in packages that include analytics calls. It is a necessary dependency to extract unless we want duplicate analytics code all over the place.

#### Changes proposed in this Pull Request:
* Add `@automattic/jetpack-analytics` package

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

Nothing should change. `jetpack build plugins/jetpack` and confirm that analytics events are firing without errors.

1. Open the developer console, execute the following code: `localStorage.setItem('debug', 'dops:analytics');`
2. Connect Jetpack.
3. Go to "Jetpack -> Settings", click on the tabs. Confirm that the developer console displays the events `jetpack_wpa_click` and `jetpack_wpa_page_view`.
4. Click on the "Search" icon in the top right corner ("Search for a Jetpack feature"). Confirm that you see the console event logged: `Recording Event ~ ...`.
